ONSdigital/design-system
# Simplify button macro

### What is the context of this PR?
Simplifies the button component macro by consolidating multiple options into `variants`. 
 
Fixes #2128 , fixes #2047 , fixes #2194 and fixes #2418 

Also fixes some styling issues with button variants to ensure icons display correctly across all variants.

This precedes #2048 which will simplify the component even further.

### What are the breaking changes?

1. Developers using any of the following parameters in either the button component,`submitButton` in the question pattern, or `signOutButton` in the header, will need to replace them with `variants` set with a string or an array of strings.

- `buttonStyle` --> e.g. `"variants": ['secondary', 'print']`
- `submitType` --> e.g. `"variants": 'loader'`
- `disabled` (attribute) --> e.g. `"variants": ['small', 'disabled']`

2. The Developers using the `signOutButton` in the header component can no longer change the type or position of the button's icon. `iconType` and `iconPosition` parameters have been removed to ensure strict usage of the sign out button.

### Additional changes
Removes the dead links to the recently unpublished 'experimental' pattern for download resources.

### How to review
- Check all the button variants produce the same html output as previous to this change
- Check all button variants render correctly
- Check tests pass
